horizon/static/framework/conf/resource-type-registry.service.js

```diff
@@ -477,13 +483,16 @@
      * If a configuration is supplied, the resource type is extended to
      * use the configuration's properties.
      */
-    function getResourceType(type, config) {
+    function getResourceType(type, config, slug) {
```

Comment:
like you said...this is awkward as an option param on a getter. Makes more sense as a setter.